app/forms/publish_interface/about_your_organisation_form.rb

```diff
@@ -0,0 +1,92 @@
+module PublishInterface
+  class AboutYourOrganisationForm
+    include ActiveModel::Model
+    include ActiveModel::Validations
+
+    attr_accessor :provider
+    attr_accessor(
+      :provider_code, :recruitment_cycle_year, :provider_name,
+      :train_with_us, :train_with_disability,
+      :email, :telephone, :urn, :website, :ukprn, :address1, :address2,
+      :address3, :address4, :postcode, :region_code
+    )
+
+    def self.build_from_provider(provider)
+      new(
```

Comment:
adding to this https://github.com/DFE-Digital/teacher-training-api/pull/2340#discussion_r770526301 and an alternative,  you could probably just do something like `super(provider.attributes.slice(*FIELDS)` in `#new` and pass the provider in when you do `AboutYourOrganisationForm.new(provider)`

Reply:
In order to address the [bug](https://github.com/DFE-Digital/teacher-training-api/pull/2340#pullrequestreview-833255533) identified by @defong  I think we need slightly different logic when building the form with the persisted provider vs building it with user-submitted params. I've opted to have two different build methods to accommodate this. WDYT?

Reply:
I've got a PR to simplify it even further. Have a look at this https://github.com/DFE-Digital/teacher-training-api/pull/2354. It uses a pattern we developed at Register as we already solved these types of issues. Happy to discuss the approach

Reply:
@darokel looks good. This PR is getting unwieldy — I suggest we card up refactoring to that pattern, and get this merged as-is (pending approval for current set of changes).